packages/circuit-ui/components/Modal/Modal.stories.tsx

```diff
@@ -13,170 +13,160 @@
-    {({ onClose }) => (
-      <ModalWrapper>
-        <ModalHeader title="A modal" />
-        <Body>Some text in the modal body.</Body>
-        <ModalFooter>
-          <ButtonGroup>
-            <Button
-              variant="secondary"
-              onClick={(event: MouseEvent | KeyboardEvent) => {
-                action('Cancel button clicked')(event);
-                onClose(event);
-              }}
-            >
-              Cancel
-            </Button>
-            <Button
-              variant="primary"
-              onClick={(event: MouseEvent | KeyboardEvent) => {
-                action('Confirm button clicked')(event);
-                onClose(event);
-              }}
-            >
-              Confirm
-            </Button>
-          </ButtonGroup>
-        </ModalFooter>
-      </ModalWrapper>
-    )}
-  </PageWithModal>
-);
-
-export const WithCustomStyles = (args: ModalProps) => {
-  const Container = styled('div')`
-    display: flex;
-    justify-content: stretch;
-    align-items: stretch;
-    flex-wrap: nowrap;
-    height: 100%;
-    background: #fff;
-  `;
-
-  const LeftColumn = styled('div')`
-    display: flex;
-    align-items: center;
-    width: 50%;
-    justify-content: center;
-    padding: 24px 18px;
-  `;
-
-  const RightColumn = styled('div')`
-    height: 100%;
-    width: 50%;
-    background: no-repeat center / cover
-      url('https://source.unsplash.com/S4W2AU0t3lw/900x1600');
-  `;
+Variants.args = {
+  children: 'Hello World!',
+};
+
+export const PreventClose = (modal: ModalProps): JSX.Element => {
+  const ComponentWithModal = () => {
+    const { setModal } = useModal();
+    return (
+      <Button type="button" onClick={() => setModal(modal)}>
+        Open modal
+      </Button>
+    );
+  };
 
   return (
-    <PageWithModal
-      {...args}
-      {...defaultModal}
-      css={css`
-        padding: 0;
-        height: 50vh;
-      `}
-    >
-      {() => (
-        <div
-          css={css`
-            padding: 0;
-            height: 50vh;
-          `}
-        >
-          <Container>
-            <LeftColumn>
-              <Body>A nice custom modal for special cases.</Body>
-            </LeftColumn>
-            <RightColumn />
-          </Container>
-        </div>
-      )}
-    </PageWithModal>
+    <ModalProvider>
+      <ComponentWithModal />
+    </ModalProvider>
   );
 };
+
+PreventClose.args = {
+  children: ({ onClose }: { onClose: ModalProps['onClose'] }) => (
+    <Fragment>
+      <Body>
+        Users have to complete the action inside the modal to close it. The
+        close button is hidden and clicking outside the modal or pressing the
+        escape key does not close the modal either.
+      </Body>
+      <Button variant="primary" onClick={onClose}>
+        Close modal
+      </Button>
+    </Fragment>
+  ),
+  variant: 'immersive',
+  preventClose: true,
+};
+
+export const InitiallyOpen = (modal: ModalProps): JSX.Element => {
```

Comment:
Just noticed that this one is also not rendering the close button anymore because it doesn't have a `closeButtonLabel` (cf. Chromatic, you probably saw it too but just in case 😛)